### DISCUSSION TITLE
WSL 2

### DISCUSSION THREAD

**Discussion by DenisDelinger** [Cyfrin/foundry-full-course-cu#891] [General]
I am unfamiliar with this linux stuff...so I'm following lesson 6 and have installed wsl and set up vscode with it, but vscode is telling me i should use wsl 2. When searching about wsl 2, I get these convoluted and unintuitive instructions and information

So is this something that I should use, does it make a significant difference, and, how do you even install it?

**Comment by IAm0xAxe** [comment]


**Reply by DenisDelinger** [reply]
Thanks but of course I know this. It doesn't answer any of my questions. I would like someone who is more knowledgeable about this to help:
**Should I** proceed with wsl 2, does it matter if i stick to wsl 1, & what difference does it make,
And any useful links and such would be appreciated, since as I stated in the post the information on this is pretty convoluted online.

**Comment by theirrationalone** [comment]
wsl2 should be default. But let's apart arg.... 

cmds to set default wsl v... :

```powershell
wsl --set-default-version 1
```

Or

```powershell
wsl --set-default-version 2
```

**Reply by DenisDelinger** [reply]
I managed to upgrade finally =) And of course, it wasn't as simple as that. It took constant trial and error on every step of the process, too much for me to remember and share here. For anyone else stuck on wsl1, give this info to chatgpt: https://learn.microsoft.com/en-us/windows/wsl/install , and ask for a summary on how to upgrade to wsl2
Depending on your machine you might be able to just go through each step and it will be done relatively easily
Otherwise you'll have to go through trial and error like me, remember to ask chatgpt for solutions for each specific step that doesn't work

**Reply by MOUNIRSLAIM** [reply]
Thank you, my friend. I was searching for the solution for three days and did not find anything
